src/main/kotlin/com/jetbrains/snakecharm/inspections/SmkShadowSettingsInspection.kt

```diff
@@ -21,12 +21,12 @@ class SmkShadowSettingsInspection : SnakemakeInspection()  {
                 return
             }
 
-            val stringLiteral = st.argumentList?.arguments?.get(0)
+            val argument = if (st.argumentList?.arguments?.size != 0) st.argumentList!!.arguments[0] else null
```

Comment:
There are couple of useful methods like `st.argumentList?.arguments?.isEmpty()`, `isNotEmpty`, `isNullOrEmpty`.

Also is better to use `st.argumentList?.arguments?.firstOrNull()` it is much  shorter